float64 x
float64 y

float64 xhat
float64 yhat

float64 xdot
float64 ydot
